### ISSUE TITLE
Newline behavior horrifically broken.

### ISSUE THREAD

**Issue by ShimmerFairy** [haskell/haskell-mode#1711]
I have my emacs set up so that hitting the enter key only inserts a newline, and nothing else, but I find that hitting enter in haskell-mode not only inserts a newline, but some amount of indentation as well. Even hitting `C-j` (which as I recall should do the right thing where enter doesn't) inserts whitespace (but a different amount of it compared to enter) after the newly-inserted newline.

Please fix this so that haskell-mode stops thinking it knows better than the user's existing setup. Fancier behavior should be hidden behind a different key combination or enabled through a customize option, neither of which is the case here.

**Comment by ShimmerFairy** [comment]
I've found that putting

```
(define-key haskell-indentation-mode-map (kbd "RET") nil)
```

In my .emacs file will get back the proper behavior. But I've noticed that on a new line I have to hit `TAB` twice to get the cursor in the proper column. The first `TAB` just indents two spaces, and the second will finally put in enough whitespace for the next line of the function. Most other major modes only require one `TAB` to move the line to the right column, it's not obvious to me why there's a "only indent two spaces" option the key can take in the first place.
